model/device.go

```diff
@@ -183,6 +185,26 @@ func (gn GroupName) String() string {
 	return string(gn)
 }
 
+func (gn GroupName) Validate() error {
+	if len(gn) > 1024 {
+		return errors.New(
+			"Group name can at most have 1024 characters",
+		)
+	} else if len(gn) == 0 {
+		return errors.New(
+			"Group name cannot be blank",
+		)
+	} else if ok, _ := regexp.MatchString(
+		"^[A-Za-z0-9_-]*$", string(gn),
```

Comment:
maybe we could compile that regex, since it does not change through-out the life of the service?

Reply:
Fixed!